File: android/app/src/main/java/com/propertyfinder1/util/IntervalNode.java
package com.propertyfinder1.util;

public class IntervalNode {
    private Interval interval;
    private IntervalNode left, right;

    public IntervalNode(Interval interval) {
        this.interval = interval;
        this.left = null;
        this.right = null;
    }

    public Interval getInterval() {
        return interval;
    }

    public IntervalNode getLeft() {
        return left;
    }

    public IntervalNode getRight() {
        return right;
    }

    public void setInterval(Interval interval) {
        this.interval = interval;
    }

    public void setLeft(IntervalNode left) {
        this.left = left;
    }

    public void setRight(IntervalNode right) {
        this.right = right;
    }
}

File: ios/IntervalNode.swift
//
//  IntervalNode.swift
//  PropertyFinder1
//
//  Created by K Arun Kumar Reddy on 19/12/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

class IntervalNode {
    private var interval: Interval;
    private var left: IntervalNode?;
    private var right: IntervalNode?;
    
    init(interval: Interval) {
        self.interval = interval;
        self.left = nil;
        self.right = nil;
    }
    
    func getInterval() -> Interval {
        return self.interval;
    }
    
    func getLeft() -> IntervalNode? {
        return self.left;
    }
    
    func getRight() -> IntervalNode? {
        return self.right;
    }
    
    func setInterval(interval: Interval) {
        self.interval = interval;
    }
    
    func setLeft(node: IntervalNode?) {
        self.left = node;
    }
    
    func setRight(node: IntervalNode?) {
        self.right = node;
    }
}
